### PR TITLE
refactor: Update Year Date in Footer.tsx

### LINKED ISSUE #455: Bug: Copyright year should be updated
### Describe the bug

In the footer of this site, we can see that the year is 2022
image
which needs to be updated 

and also it should be automated to change automatically when the year changes

### Steps to reproduce

1. Go to https://hot.opensauced.pizza/
2. Scroll till the end 
3. Notice that the year beside the copyright symbol needs to be updated

### Affected services

hot.opensauced.pizza

### Platforms

Desktop, Mobile, Tablet, TV

### Browsers

Chrome, Firefox, Safari, Edge, Opera

### Environment

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by tmsagarofficial**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR fixes #455 
I wanted to get help too on this I could have changed the year fro 2022 to 2023 but I am making it automatic .. 
## Related Tickets & Documents
```
document.write(new Date().getFullYear())
```
I have added the following as a replacement to the static year 2022 .
but should I have added this below code??
```
console.log(document.write(new Date().getFullYear()))
```

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/components/Footer.tsx [hunk 1]
```diff
@@ -146,7 +146,7 @@ const Footer = (): JSX.Element => (
 
         <div className="flex md:justify-center lg:hidden lg:border-none lg:order-2 border-t py-3 pb-4 mt-2 text-sm justify-between">
           <p className="text-light-slate-9">
-            © 2022
+            document.write(new Date().getFullYear())
```
[the remark this answers is not in the document]
Comment (NsdHSO) [reply src/components/Footer.tsx:149]: I see that before it was ©. is this a feature?

**Comment by NsdHSO** [comment]
HI @tmsagarofficial  I will make a change in the title.

File: src/components/Footer.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (TAKANOME-DEV) [reply src/components/Footer.tsx:149]: > I see that before it was ©.
>
I definitely forget about `©` :sweat_smile: 
> is this a feature?
>
A bug fix

**Review by bdougie — APPROVED** [review]
this works

**Comment by bdougie** [comment]
> @bdougie What do I do now? what about the tests which have failed? 😅

`npm run format`

**Review by TAKANOME-DEV — APPROVED** [review]
Looks Good :+1: Just run the `format` script and push again :pizza:

**Comment by 0-vortex** [comment]
> Is it done now?

You have to change the values that `npm run format` is telling you it can't fix, then it is ready! 😅 🍕

**Comment by 0-vortex** [comment]
> > > @bdougie What do I do now? what about the tests which have failed? 😅
> > 
> > 
> > `npm run format`
> > Those are lint changes needed. This will also need another approval to get merged.
> 
> I really need help with this. Where do I run this ? Also, I'm getting errors if I'm trying to run that locally

Run `npm run format` in the project folder, where you installed the dependencies.

**Comment by bdougie** [comment]
@tmsagarofficial, @0-vortex & @TAKANOME-DEV but I think there is a disconnect in your knowledge on how [npm run format](https://github.com/open-sauced/hot/blob/main/package.json#L49) works. 

Perhaps leverage the [Discord](https://discord.gg/opensauced) for clarification. 

@diivi if you are looking for a new [docs](https://docs.opensauced.pizza/) addition. `npm run format` would be a great addition.

**Comment by diivi** [comment]
Nice,
Maybe I can do something like this - https://torsion.org/borgmatic/docs/how-to/develop-on-borgmatic/#code-style (from another project I contribute to).
Does OpenSauced have a Code Style guide or something related that I could refer to?

**Comment by diivi** [comment]
Okay, just checked the `README` for this project; a new heading `Code Linting`  in the `Introduction to Contributing Section` sounds good.

**Comment by bdougie** [comment]
> Okay, just checked the `README` for this project; a new heading `Code Linting` in the `Introduction to Contributing` section sounds good.

Better title, "Applying Lint Styleguide"

**Comment by diivi** [comment]
What do you think about using git hooks/pre-commit or husky for these tasks too?

**Comment by TAKANOME-DEV** [comment]
> Let's see now

@tmsagarofficial follow these steps and it should be good:
1. In the root of the project, run `npm run format`
2. Next, build the project to make sure it's all good: `npm run build`
3. Add the changes: `git add src/components/Footer.tsx`
4. Commit and push :pizza:

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -72,8 +72,10 @@ const Footer = (): JSX.Element => (
           <span className="lg:hidden font-bold text-light-slate-12 ">OpenSauced</span>
 
           <p className="hidden !text-light-slate-9 lg:inline-block">
-            © 2022
-            {" "}
+           ©
```
Comment (diivi) [inline src/components/Footer.tsx:75]: nit: `&copy`

**Comment by 0-vortex** [comment]
> What do you think about using husky for these tasks too? I can try to set one up for the format, test scripts. also checking semantic PRs client side could be easier with husky.

Added 

> > Let's see now
> 
> @tmsagarofficial follow these steps and it should be good:
> 
> 1. In the root of the project, run `npm run format`
> 2. Next, build the project to make sure it's all good: `npm run build`
> 3. Add the changes: `git add src/components/Footer.tsx`
> 4. Commit and push 🍕

@TAKANOME-DEV 4 should be: `npm run push` instead of commit if you want to really follow the docs!!!

**Comment by 0-vortex** [comment]
@diivi @bdougie There is a work in progress on what the correct style guide for the frontend is, both feeding off the backend rules but not consolidated. While hot and insights have diverged in that manner, we did not have the time or resources to unify the rules and only apply the generic `npm run format` for eslint fixing in a holistic way. The reason it's somewhat performant is that our configuration is currently made up only of auto-format rules, meaning running lint --fix would __generally__ solve most issues.

Because there are multiple ways a user can overload this, like the OP is potentially using IDE formatting, we should only document how to use them and potential bottlenecks. Part of the triage/maintainer docs on this are already live, while the introduction to contributing doesn't go into much detail. Would love sections like "how to open your first issue" or "how to open your first commit" but feel like that doesn't solve OP problem :<

**Comment by bdougie** [comment]
> There is a work in progress on what the correct style guide for the frontend is, both feeding off the backend rules but not consolidated. While hot and insights have diverged in that manner, we did not have the time or resources to unify the rules and only apply the generic npm run format for eslint fixing in a holistic way. 

Adding reference to npm run format is all that is needed in the docs. We don't need to list out the style guide.

**Pull request closed without merging**